src/dataExplorer/components/ResultsContext.tsx

```diff
@@ -9,25 +9,57 @@ interface ResultsContextType {
 
   setStatus: (status: RemoteDataState) => void
   setResult: (result: FluxResult) => void
-  setTime: (time: number) => void
 }
 
 export const ResultsContext = createContext<ResultsContextType>({
   status: RemoteDataState.NotStarted,
   result: {} as FluxResult,
-  time: 0,
+  time: null,
 
   setStatus: _ => {},
   setResult: _ => {},
-  setTime: _ => {},
 })
 
 export const ResultsProvider: FC = ({children}) => {
   const [result, setResult] = useState<FluxResult>({} as FluxResult)
-  const [time, setTime] = useState<number>(0)
+  const timeStart = useRef<number>(null)
+  const [time, setTime] = useState<number>(null)
   const [status, setStatus] = useState<RemoteDataState>(
     RemoteDataState.NotStarted
   )
+
+  useEffect(() => {
+    let running = false
+    const check = () => {
+      if (!running) {
+        return
+      }
+
+      setTime(Date.now() - timeStart.current)
+
+      window.requestAnimationFrame(check)
```

Comment:
so `check` is calling itself recursively until the status gets updated to non-loading?

Reply:
yes